### PR TITLE
Closes #60

### LINKED ISSUE #60: Update metrics.py, esquery.py and test_esquery.py to use only elasticsearch_dsl module
After #57, this issue adds the use of elasticsearch_dsl to other files such as metrics.py.
 
`esquery.py` in the function `get_aggs`, returns a json dict which is then used in metrics.py to get the results via `requests.get` method. 
This issue is about removing completely the direct querying to elasticsearch and using elasticsearch_dsl.py instead.

EDIT: The second PR is ready, Please review #58 so that the second PR can be added on top of it.

### PR DESCRIPTION
**Pull request by aswanipranjal**
Updated comments in esquery.py.
Metrics.py uses elasticsearch_dsl.py instead of directly requesting easticsearch for data.
Updated test for get_agg.

### CODE REVIEW

File: manuscripts/esquery.py [hunk 1]
```diff
@@ -241,9 +241,11 @@ def __get_bounds(cls, start=None, end=None):
             start_ts = None
             end_ts = None
             if start:
+                start = start.replace(microsecond=0)
                 start_ts = start.replace(tzinfo=timezone.utc).timestamp()
                 start_ts_ms = start_ts * 1000  # ES uses ms
             if end:
+                end = end.replace(microsecond=0)
```
Comment (acs) [inline manuscripts/esquery.py]: Why do we need to set microsecond to 0? Is not configured in this way by default?

File: manuscripts/metrics/metrics.py [hunk 2]
```diff
@@ -157,12 +157,17 @@ def get_metrics_data(self, query):
...
-        return r.json()
+            url = 'http://' + self.es_url
+        es = Elasticsearch(url)
+        s = Search(using=es, index=self.es_index)
+        s = s.update_from_dict(query)
+        try:
+            response = s.execute()
+            return response.to_dict()
+        except Exception as e:
+            raise e
```
Comment (acs) [inline manuscripts/metrics/metrics.py]: Capturing the Exception an reraising it without doing any thing is a bit weird. And having the return inside the try also.

File: manuscripts/esquery.py (hunk 1 above)
Comment (aswanipranjal) [reply manuscripts/esquery.py]: @acs: No, it isn't configured this way by default and results in an error from elasticsearch.
screen shot 2018-06-04 at 11 46 57 am
This error presents it self when we convert datetime to timestamp.
I'll update the file and mention it there.

File: manuscripts/metrics/metrics.py (hunk 2 above)
Comment (aswanipranjal) [reply manuscripts/metrics/metrics.py]: My bad. I wanted to use the error and show the exact information about the error to the user to make debugging easier.

File: manuscripts/esquery.py (hunk 1 above)
Comment (acs) [reply manuscripts/esquery.py]: Ah, ok, it is related with the migration to elasticsearch_dsl. Great! I want just to understand the need for it. Thanks you @aswanipranjal ! LGTM

File: manuscripts/metrics/metrics.py (hunk 2 above)
Comment (acs) [reply manuscripts/metrics/metrics.py]: IMHO once this comment is addressed the PR is ready to be merged.

**Comment by aswanipranjal** [comment]
PR #64 and this one have similar errors. I am looking into them.

**Comment by aswanipranjal** [comment]
@jgbarah, @acs, should I change the function `get_query_filters` and then rewrite the tests? Right now, as far as I can tell, the dictionaries are identical but the lists inside them might have same items in different orders.

**Comment by jgbarah** [comment]
I see that fixed the test. So, ready to go!

**Pull request merged**